From fa30b5c46a151d53e17490ca179d833db986b6b5 Mon Sep 17 00:00:00 2001
From: Charles Ci-Jyun Wu <dminus@andestech.com>
Date: Thu, 14 Sep 2023 21:24:21 +0800
Subject: [PATCH 023/168] riscv: andes: add andes vendor sbi fid to
 include/linux/soc/andes/sbi.h

Andes has defined SBI function IDs within the vendor ID architecture.

Signed-off-by: Charles Ci-Jyun Wu <dminus@andestech.com>
---
 arch/riscv/errata/andes/errata.c |  9 +++----
 include/linux/soc/andes/sbi.h    | 43 ++++++++++++++++++++++++++++++++
 2 files changed, 46 insertions(+), 6 deletions(-)
 create mode 100644 include/linux/soc/andes/sbi.h

diff --git a/arch/riscv/errata/andes/errata.c b/arch/riscv/errata/andes/errata.c
index f2708a9494a1..b1627a1cabd4 100644
--- a/arch/riscv/errata/andes/errata.c
+++ b/arch/riscv/errata/andes/errata.c
@@ -15,24 +15,21 @@
 #include <asm/errata_list.h>
 #include <asm/patch.h>
 #include <asm/processor.h>
-#include <asm/sbi.h>
 #include <asm/vendorid_list.h>
+#include <linux/soc/andes/sbi.h>
 
 #define ANDES_AX45MP_MARCHID		0x8000000000008a45UL
 #define ANDES_AX45MP_MIMPID		0x500UL
-#define ANDES_SBI_EXT_ANDES		0x0900031E
-
-#define ANDES_SBI_EXT_IOCP_SW_WORKAROUND	1
 
 static long ax45mp_iocp_sw_workaround(void)
 {
 	struct sbiret ret;
 
 	/*
-	 * ANDES_SBI_EXT_IOCP_SW_WORKAROUND SBI EXT checks if the IOCP is missing and
+	 * SBI_EXT_ANDES_IOCP_SW_WORKAROUND SBI EXT checks if the IOCP is missing and
 	 * cache is controllable only then CMO will be applied to the platform.
 	 */
-	ret = sbi_ecall(ANDES_SBI_EXT_ANDES, ANDES_SBI_EXT_IOCP_SW_WORKAROUND,
+	ret = sbi_ecall(ANDES_SBI_EXT_ANDES, SBI_EXT_ANDES_IOCP_SW_WORKAROUND,
 			0, 0, 0, 0, 0, 0);
 
 	return ret.error ? 0 : ret.value;
diff --git a/include/linux/soc/andes/sbi.h b/include/linux/soc/andes/sbi.h
new file mode 100644
index 000000000000..d21ba2cc5819
--- /dev/null
+++ b/include/linux/soc/andes/sbi.h
@@ -0,0 +1,43 @@
+/* SPDX-License-Identifier: GPL-2.0 */
+/*
+ * Copyright (C) 2023 Andes Technology Corporation.
+ */
+#ifndef __LINUX_SOC_ANDES_SBI_H
+#define __LINUX_SOC_ANDES_SBI_H
+
+#include <asm/sbi.h>
+#include <asm/vendorid_list.h>
+
+#define ANDES_SBI_EXT_ANDES	0x0900031E
+
+/*
+ * Set Andes's internal SBI Call FID start at 64 to avoid the impact of
+ * upstream add or remove in sbi_ext_andes_fid{}. If the FID numbers
+ * change, our released OpenSBI will lose backward compatible.
+ */
+#define ANDES_SBI_INTERNAL_FID_START 64
+
+enum sbi_ext_andes_fid {
+	SBI_EXT_ANDES_FID0 = 0, /* Reserved for future use */
+	SBI_EXT_ANDES_IOCP_SW_WORKAROUND,
+
+	/* Trace */
+	SBI_EXT_ANDES_TRIGGER_SET = ANDES_SBI_INTERNAL_FID_START,
+
+	/* CPU freq */
+	SBI_EXT_ANDES_POWERBRAKE_READ,
+	SBI_EXT_ANDES_POWERBRAKE_WRITE,
+
+	/* Programmable physical memory attributes (PPMA) */
+	SBI_EXT_ANDES_PMA_SET,
+	SBI_EXT_ANDES_PMA_FREE,
+	SBI_EXT_ANDES_PMA_PROBE,
+
+	SBI_EXT_ANDES_DCACHE_EN,
+
+	/* CPU idle (ATCSMU) */
+	SBI_EXT_ANDES_SUSPEND_MODE_SET,
+	SBI_EXT_ANDES_SUSPEND_MODE_ENTER,
+};
+
+#endif /* !__LINUX_SOC_ANDES_SBI_H */
-- 
2.34.1

